pulsar-client-cpp/include/pulsar/MessageRoutingPolicy.h

```diff
@@ -33,7 +34,7 @@ class MessageRoutingPolicy {
  public:
     virtual ~MessageRoutingPolicy() {}
 
-    virtual int getPartition(const Message& msg) = 0;
+    virtual int getPartition(const Message& msg, const TopicMetadata& topicMetadata) = 0;
```

Comment:
The original method needs to be there as since it's part of the user API. We could mark it as deprecated and remove it in 2.0. 

If the user was defining a custom router, it should continue working after this change

Reply:
@merlimat We need to set these method to `virtual` because of the restriction of Google Mock.
https://github.com/google/googletest/blob/master/googlemock/docs/ForDummies.md#how-to-define-it